lib/foreman/renderer.rb

```diff
@@ -107,7 +120,9 @@ def snippet(name, options = {})
         begin
           return unattended_render(template, nil, options[:variables] || {})
         rescue => exc
-          raise "The snippet '#{name}' threw an error: #{exc}"
+          e = ::Foreman::Exception.new(N_("The snippet '%{name}' threw an error: %{exc}"), { :name => name, :exc => exc })
```

Comment:
I'm sorry I fail to locate the `rescue` in `unattended_render` you mention, could you please point me to it so I could better understand the change? If you meant the nested in in `safe_render` then it only catches syntax errors. If that would be other exception, e.g. NoMethodError when nil does not respond to something, it would not be rescued properly. Maybe we could skip the wrapping of exception and simply re-raise it if `exc` is kind of `Foreman::Exception`, in such case we know it was already handled by us.